### ISSUE TITLE
`:` doesn't need to be quoted in a value

### ISSUE THREAD

**Issue by kornelski** [chyh1990/yaml-rust#118]
This parser is unable to parse this file:

https://github.com/sublimehq/Packages/blob/d5124e6c195befc1af0cb67e0acfa29670c1dbd5/Go/Go.sublime-syntax#L256

because of `match: :=` which is treated as a syntax error rather than unquoted. This makes [`syntect`](https://crates.rs/crates/syntect) fail to parse its syntaxes.

**Comment by robinst** [comment]
YAML 1.2 spec about this, section [7.3.3. Plain Style](https://yaml.org/spec/1.2/spec.html#id2788859):

> However, the “:”, “?” and “-” indicators may be used as the first character if followed by a non-space “safe” character, as this causes no ambiguity.

**Comment by robinst** [comment]
If anyone is keen to look into it, here's a failing test case for this problem (add to [tests/spec_test.rs](https://github.com/chyh1990/yaml-rust/blob/master/tests/spec_test.rs)):

```
#[test]
fn test_plain_scalar_starting_with_colon_in_flow() {
    use yaml_rust::{Yaml, YamlLoader};
    use yaml_rust::yaml::Hash;

    let source = "{x: :=}";
    let documents = YamlLoader::load_from_str(&source).unwrap();
    assert_eq!(documents.len(), 1);

    let mut hash = Hash::new();
    hash.insert(Yaml::String("x".to_owned()), Yaml::String(":=".to_owned()));
    let expected = Yaml::Hash(hash);
    assert_eq!(documents[0], expected);
}
```

**Comment by robinst** [comment]
This was tricky, but here's a PR to fix this: #122
